include/vara/Feature/FeatureModel.h

```diff
@@ -154,28 +153,29 @@ class FeatureModelBuilder : private FeatureModel {
     Children.clear();
   }
 
-  /// Try to create and add a new \a Feature to the \a FeatureModel.
+  /// Try to create a new \a Feature.
   ///
   /// \param[in] FeatureName name of the \a Feature
   /// \param[in] FurtherArgs further arguments that should be passed to the
   ///                        \a Feature constructor
   ///
-  /// \returns true, if the feature could be inserted into the \a FeatureModel
+  /// \returns ptr to inserted \a Feature
   template <typename FeatureTy, typename... Args,
             typename = typename std::enable_if_t<
                 std::is_base_of_v<Feature, FeatureTy>, int>>
-  bool makeFeature(const std::string &FeatureName, Args... FurtherArgs) {
-    // TODO: maybe this should return the created Feature
-    return Features
-        .try_emplace(FeatureName, std::make_unique<FeatureTy>(
-                                      FeatureName, std::move(FurtherArgs)...))
-        .second;
+  Feature *makeFeature(const std::string &FeatureName, Args... FurtherArgs) {
+    if (!Features
+             .try_emplace(FeatureName,
+                          std::make_unique<FeatureTy>(
+                              FeatureName, std::move(FurtherArgs)...))
+             .second) {
+      return nullptr;
+    }
+    return Features[FeatureName].get();
   }
 
-  bool addFeature(Feature &F);
-
-  FeatureModelBuilder *addParent(const std::string &FeatureName,
-                                 const std::string &ParentName) {
+  FeatureModelBuilder *addEdge(const std::string &ParentName,
```

Comment:
Why did you rename this? `addParent` seemed fine to me

Reply:
I wanted to clarify, that calling this implies, that the children relation between these nodes will be updated too (while building the tree) -- in contrast to the `FeatureModelModification` where we implemented `setParent` so it literally changes only that.